File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    maxBig = 0
    maxSmall = 0
    for [h, w] in sizes:
        big = h if h>w else w
        small = w if h>w else h
        
        maxBig = big if big > maxBig else maxBig
        maxSmall = small if small > maxSmall else maxSmall
    return maxBig * maxSmall

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;
import java.util.stream.*;
class Solution {
    public int solution(int[][] sizes) {
        int w = 0;
        int h = 0;
        for(int i = 0; i<sizes.length; i++){
            int[] size = sizes[i];
            w = Math.max(Math.max(size[0],size[1]), w);
            h = Math.max(Math.min(size[0],size[1]), h);
        }
        return w*h;
    }
}